lesson18.md

```diff
@@ -0,0 +1,385 @@
+<!-- .element: class="fragment" -->
+
+
+---
+
+### Bookshelf array
+
+![bookshelf](images/array_bookshelf_0.png)
+
+---
+
+### Invalid elements
+
+```js
+let books = ["Harry Potter", "The Hobbit", "Game of Thrones"];
+
+console.log(books[0]); // "Harry Potter"
+console.log(books[1]); // "The Hobbit"
+console.log(books[2]); // "Game of Thrones"
+console.log(books[3]); // ???
+```
+
+```js
+console.log(books[3]); // undefined
+```
+<!-- .element: class="fragment" -->
+
+---
+
+### Quiz
+
+```js
+let friends = ["Alice", "Bob", "Carol"];
+
+console.log(friends[1]); // ???
+console.log(friends[3]); // ???
+```
+
+```js
+console.log(friends[1]); // "Bob"
+console.log(friends[3]); // undefined
+```
+<!-- .element: class="fragment" -->
+
+
+
+### Exercise 1
+
+* Step 1: Create an array with your 3 top friends
+* Step 2: Say "hello" on console to each friend, e.g.:
+
+```plaintext
+hello Alice
+hello Bob
+hello Carol
+```
+
+---
+
+### Solution
+
+```js
+let friends = ["Sevtap", "Carlo", "Abdullah"];
+console.log("hello " + friends[0]);
+console.log("hello " + friends[1]);
+console.log("hello " + friends[2]);
+```
+
+---
+
+### Modifying arrays
+
+We can change any value using brackets:
+
+```js
+let friends = ["Alice", "Bob", "Carol"];
+
+friends[1] = "David";
+// friends array is now ["Alice", "David", "Carol"]
+
+console.log(friends[1]); // "David"
+```
+
+---
+
+### Common operations: array length
+
+We can get the length of an array with the `.length` property:
+
+```js
+let friends = ["Alice", "Bob", "Carol"];
+console.log(friends.length); // 3
+```
+
+---
+
+### Iterating through an array
+
+How can we make the code below more generic, for any array size?
+
+```js
+let cars = ["Saab", "Volvo", "BMW"];
+
+console.log(cars[0]);
+console.log(cars[1]);
+console.log(cars[2]);
+```
+
+---
+
+### Loops!
+
+```js
+let cars = ["Saab", "Volvo", "BMW"];
+
+for (let i = 0; i < cars.length; i++) {
+    console.log(cars[i]);
+}
+```
+
+---
+
+### Exercise 2
+
+* Step 1: Create an array with your 3 top friends (you can use the same from exercise 1)
+* Step 2: Say "hello" on console to each friend, using a `for` loop!
+
+```
+hello Alice
+hello Bob
+hello Carol
+```
+
+---
+
+### Appending new values
+
+We can append new values to an array using `.push()`:
+
+```js
+let friends = ["Alice", "Bob", "Carol"];
+console.log(friends.length); // 3
+
+friends.push("David");
+
+console.log(friends.length); // 4
+console.log(friends[3]); // David
+```
+
+---
+
+### Exercise 3
+
+Create a function that takes an array and outputs the following:
+
+* Output the last element of the array
+* Output the sum of all the numbers in the array
+* Output the largest number of the array
+* Output the average (mean) of all numbers (sum of all numbers divided by amount of numbers)
+
+Create an array containing a few numbers, ex `[3, 2, 9, 5, 6]` and test your code:
+
+```js
+function exercise3(array) {
+    // your code here
+}
+exercise3([3, 2, 9, 5, 6]);
```

Comment:
IMO, when we teach new syntax we should keep things as simple as possible. Some students have difficulty understanding inlined arguments:

```suggestion
let myNumbers = [3, 2, 9, 5, 6];
exercise3(myNumbers);
```

Reply:
yes, completely agree, totally overlooked this